File: app/src/main/java/com/example/appadm/FormLogin.java
package com.example.appadm;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.lifecycle.ViewModelProvider;

import com.example.appadm.ViewModel.FormLoginViewModel;
import com.google.android.material.snackbar.Snackbar;

public class FormLogin extends AppCompatActivity {

    private EditText editTextEmail;
    private EditText editTextPassword;
    private FormLoginViewModel formLoginViewModel;
    private Button button_criar_conta;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_form_login);

        // Habilita o uso da biblioteca EdgeToEdge para ajustar as margens conforme a barra de status e navegação
        EdgeToEdge.enable(this);

        Log.i("LOG", "onCreate() FormLogin");
        // Define o listener de insets para ajustar as margens conforme a barra de status e navegação
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.formlogin), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // Inicializa o botão de criar conta
        button_criar_conta = findViewById(R.id.button_criar_conta);

        editTextEmail = findViewById(R.id.login);
        editTextPassword = findViewById(R.id.senha);

        formLoginViewModel = new ViewModelProvider(this).get(FormLoginViewModel.class);



    }

    @Override
    protected void onStart() {
        super.onStart();

        Log.i("LOG", "onStart() FormLogin");
    }

    @Override
    protected void onResume() {
        super.onResume();

        Log.i("LOG", "onResume() FormLogin");


        button_criar_conta.setOnClickListener(v -> {
            // Cria a intent para abrir a tela de cadastro oficial
            Intent intent = new Intent(FormLogin.this, TelaCadastroOficial.class);
            startActivity(intent);
        });

        Button buttonEntrar = findViewById(R.id.button_entrar);
        buttonEntrar.setOnClickListener(v -> {
            String email = getEmailFromEditText();
            String password = getPasswordFromEditText();
            formLoginViewModel.loginUser(email, password);

        });


        formLoginViewModel.observeLoginSuccess().observe(this, loginSuccess -> {
            if (loginSuccess) {
                openTelaControle();
            } else {
                // Mostrar mensagem de erro de login
            }
        });



    }

    @Override
    protected void onPause() {
        super.onPause();

        Log.i("LOG", "onPause() FormLogin");
    }

    @Override
    protected void onStop() {
        super.onStop();

        Log.i("LOG", "onStop() FormLogin");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        Log.i("LOG", "onDestroy() FormLogin");
    }



    private void openTelaControle() {
        Intent intent = new Intent(this, TelaControle.class);
        startActivity(intent);
    }

    private String getEmailFromEditText() {
        return editTextEmail.getText().toString();
    }

    // Método para obter a senha do campo de texto
    private String getPasswordFromEditText() {
        return editTextPassword.getText().toString();
    }


}

File: app/src/main/java/com/example/appadm/ui/view/FormLogin.kt
package com.example.appadm.ui.view

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.appadm.databinding.ActivityFormLoginBinding
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException
import com.google.firebase.auth.FirebaseAuthInvalidUserException

class FormLogin : AppCompatActivity() {
    private lateinit var binding: ActivityFormLoginBinding
    private val messages = arrayOf("Preencha todos os campos", "Login realizado com sucesso")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityFormLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
        enableEdgeToEdge()
        setupWindowInsets()
        setupClickListeners()
    }

    override fun onResume() {
        super.onResume()
        // Atualizar a interface do usuário quando a atividade for retomada
        updateUI()
    }

    private fun setupWindowInsets() {
        ViewCompat.setOnApplyWindowInsetsListener(binding.formlogin) { view: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            view.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    private fun setupClickListeners() {
        binding.buttonEntrar.setOnClickListener { view ->
            val email = binding.editEmail.text.toString()
            val password = binding.editSenha.text.toString()
            if (email.isEmpty() || password.isEmpty()) {
                showSnackbar(view, messages[0])
            } else {
                loginUser(view, email, password)
            }
        }

        binding.buttonCriarConta.setOnClickListener {
            startActivity(Intent(this, FormRegister::class.java))
        }
    }

    private fun loginUser(view: View, email: String, password: String) {
        FirebaseAuth.getInstance().signInWithEmailAndPassword(email, password)
            .addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    showSnackbar(view, messages[1])
                    openControlScreen()
                } else {
                    val error = when (task.exception) {
                        is FirebaseAuthInvalidUserException -> "Usuário não encontrado."
                        is FirebaseAuthInvalidCredentialsException -> "Credenciais inválidas."
                        else -> "Erro ao realizar login."
                    }
                    showSnackbar(view, error)
                }
            }
    }

    private fun openControlScreen() {
        startActivity(Intent(this, ControlHomeScreen::class.java))
    }

    private fun showSnackbar(view: View, message: String) {
        Snackbar.make(view, message, Snackbar.LENGTH_SHORT)
            .setBackgroundTint(Color.BLUE)
            .setTextColor(Color.BLACK)
            .show()
    }

    private fun updateUI() {
        // Atualizar qualquer parte da interface do usuário que precise ser renovada quando a atividade é retomada
        // Exemplo: limpar campos de texto
        binding.editEmail.text.clear()
        binding.editSenha.text.clear()
    }
}
